File: backend/src/models/Insight.js
const mongoose = require('mongoose');
const crypto = require('crypto');

const InsightSchema = new mongoose.Schema(
  {
    insightId: {
      type: String,
      default: () => crypto.randomUUID(),
      unique: true,
      index: true
    },
    userId: {
      type: String,
      ref: 'User',
      required: [true, 'User ID is required'],
      index: true
    },
    type: {
      type: String,
      required: [true, 'Insight type is required'],
      enum: {
        values: ['spending_alert', 'savings_tip', 'budget_warning', 'monthly_summary'],
        message: '{VALUE} is not a valid insight type'
      }
    },
    message: {
      type: String,
      required: [true, 'Insight message is required'],
      trim: true
    },
    isRead: {
      type: Boolean,
      default: false
    }
  },
  {
    timestamps: true,
    toJSON: {
      transform: function (doc, ret) {
        delete ret._id;
        delete ret.__v;
        return ret;
      }
    }
  }
);

module.exports = mongoose.model('Insight', InsightSchema);

File: flutter_app/lib/models/insight.dart
class InsightModel {
  final String insightId;
  final String userId;
  final String type; // 'spending_alert', 'savings_tip', 'budget_warning', 'monthly_summary'
  final String message;
  final bool isRead;
  final DateTime? createdAt;

  InsightModel({
    required this.insightId,
    required this.userId,
    required this.type,
    required this.message,
    required this.isRead,
    this.createdAt,
  });

  factory InsightModel.fromJson(Map<String, dynamic> json) {
    return InsightModel(
      insightId: json['insightId'] ?? '',
      userId: json['userId'] ?? '',
      type: json['type'] ?? 'savings_tip',
      message: json['message'] ?? '',
      isRead: json['isRead'] ?? false,
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'insightId': insightId,
      'userId': userId,
      'type': type,
      'message': message,
      'isRead': isRead,
      'createdAt': createdAt?.toIso8601String(),
    };
  }
}
